src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -657,20 +780,22 @@ void TriggeredPublisher::DoWork()
       this->newMatchSignal.wait_for(lock, 1s,
         [this]
         {
-          return (this->publishCount > 0) || this->done;
+          return (this->publishCount > 0) || this-> triggerServ || this->done;
         });
 
       if (this->publishCount == 0 || this->done)
+      {
         continue;
+      }
 
-      std::swap(pending, this->publishCount);
-    }
-
-    for (auto &info : this->outputInfo)
-    {
-      for (std::size_t i = 0; i < pending; ++i)
+      if (this->publishCount >0)
       {
-        info.pub.Publish(*info.msgData);
+        std::swap(pending, this->publishCount);
+        PublishMsg(pending);
```

Comment:
I think this line does not need to be moved into the mutex scope. The original code had the publishing outside the scope. Unless there's a reason to change it, we shouldn't change existing behavior.